.github/workflows/linux.yml

```diff
@@ -0,0 +1,133 @@
+name: linux
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+  push:
+    branches:
+      - master
+      - TODO
+
+jobs:
+  cmake:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-16.04, ubuntu-18.04]
```

Comment:
Don't think that testing on ubuntu-16 and ubuntu-18 makes much of difference, except for increasing build time

Reply:
I will remove ubuntu-16.04

Reply:
Should I also remove windows-2016 and keep only window-2019?

Reply:
Does it affects overall build time? If yes and kind of significantly than yes, otherwise no (the thing with Linux is it just works, unlike windows, although frankly speaking I don't know much about windows and don't even bother about this)

Reply:
The limit of total concurrent jobs is 20. Too many jobs  take a long time to run tests.
So I deleted most matrix and only kept one on windows-2016.